src/groupByKey.ts

```diff
@@ -2,6 +2,14 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T, K extends keyof T>(items: T[], key: K)
+  : GroupsMap<T> {
+  return items.reduce((prev, current) => {
+    const thisKey = [current[key]];
+
+    return {
+      ...prev,
+      [thisKey as any]: items.filter((object) => object[key] === current[key]),
```

Comment:
should thisKey have any type?

Reply:
@AlisaVlasova TS shows an error then
<img width="567" alt="image" src="https://user-images.githubusercontent.com/104283893/205368718-1374e12f-84b7-43fd-9265-05c874b53594.png">
<img width="753" alt="image" src="https://user-images.githubusercontent.com/104283893/205368764-da5263e9-e8cc-44a1-a381-8d7f2701396b.png">